### ISSUE TITLE
Blog Content Block Encrypt

### ISSUE THREAD

**Issue by YunYouJun** [YunYouJun/valaxy#249]
### Clear and concise description of the problem

已支持文章加密功能，后续可考虑抽取独立组件，供用户加密文章部分内容。

### Suggested solution

- https://github.com/YunYouJun/valaxy/issues/248#issuecomment-1666916160

### Validations

- [X] Check that there isn't already an issue that request the same feature to avoid creating a duplicate.

**Comment by yixiaojiu** [comment]
How to use this component? Maybe something like this?

```md
<ValaxyDecrypt password="valaxy">
content...
</ValaxyDecrypt>
```

**Comment by YunYouJun** [comment]
You are right, this is expected API, but has not yet verified the implementation.

**Comment by yixiaojiu** [comment]
> You are right, this is expected API, but has not yet verified the implementation.

I would like to attempt to implement this feature.

**Comment by YunYouJun** [comment]
Cool! Look forward to it!
If you need any help, please let me know.

**Comment by yixiaojiu** [comment]
Is it unreasonable to use the `ValaxyEncrypt` component to encrypt parts of markdown? First of all, the markdown content wrapped by the `ValaxyEncrypt` component will be compiled by markdown-it, but the compilation result is incorrect. Secondly, the encryption work is completed on the node side, and the `ValaxyEncrypt` component belongs to the client side. It makes no sense to use the `ValaxyEncrypt` component to wrap the content to be encrypted. Maybe my understanding of the feature is not correct. If not, can you explain what the expected API looks like?

**Comment by yixiaojiu** [comment]
Is it possible to use comments to wrap the content to be encrypted?
```md
<!-- encrypt:valaxy -->
Content to be encrypted...
<!-- encrypt -->
```

**Comment by YunYouJun** [comment]
What you say makes a lot of sense. I haven't thought much about the form of the api here.

How about this:

```md
<!-- valaxy-encrypt:start -->
Content to be encrypted...
<!-- valaxy-encrypt:end -->
```

**Comment by yixiaojiu** [comment]
> What you say makes a lot of sense. I haven't thought much about the form of the api here.
> 
> How about this:
> 
> ```
> 
> Content to be encrypted...
> 
> ```

"Does partial encryption not require a separate password to be set? This method does not have a password set."

**Comment by YunYouJun** [comment]
Oh, sorry, I forgot it.

```md
<!-- valaxy-encrypt-start:your-password -->
Content to be encrypted...
<!-- valaxy-encrypt-end -->
```

I added `start` `end` and `valaxy` prefix to make it more distinct.
Prettier use `<!-- prettier-ignore-start -->` and `<!-- prettier-ignore-end -->` to ignore rules.
So I think `valaxy-encrypt-start` might be a better fit, and maybe in the future we might have more different blocks.

**Comment by yixiaojiu** [comment]
Thank you for your answer.

**Comment by YunYouJun** [comment]
Closed by #289
